gcp_airflow_foundations/base_class/dlp_table_config.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass, field
+from typing import Optional
+
+from gcp_airflow_foundations.base_class.dlp_source_config import DlpSourceConfig
+
+
+@dataclass
+class DlpTableConfig:
+    """
+    Attributes:
+    """
+    is_on: Optional[bool] = None
+    template_name_override: Optional[str] = None
+    rows_limit_percent_override: Optional[int] = None
+    min_match_count_override: Optional[int] = None
+    recurrence_override: Optional[str] = None
+
+
+    def set_source_config(self, source_config: DlpSourceConfig):
```

Comment:
@MichailParaskevopoulos @danielkulikov tried a slightly different approach here. 

Instead of passing both Source and Table config into the dlp builder, I am passing only table config which expects to have a reference to the source config. All configs (both table and source) get accesses via the table config - this makes overrides easier to manage. 

This is not super clean  though. Main issues I see
1) The source config needs to be passed in after initial creation which is error prone
2) For every new field we need to add an accessor function

Reply:
@eugenemiretsky Here you mean the `DlpTableConfig`, right?

Reply:
yep